### PR TITLE
Implement Fibonacci series function

### PR DESCRIPTION
**Pull request by madhavansu**
Fixes #24 

**Changes proposed in this PR** 

- Create an executable go program in directory 01_fib
- Write a function that prints the first n Fibonacci numbers

### CODE REVIEW

File: 01_fibonacci/fib.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+var out io.Writer = os.Stdout
+
+func main() {
+	fibonacci(10)
+}
+
+func fibonacci(n int) {
+
+	var sum int
+	var prev, cur int = 0, 1
```
Comment (anzdaddy) [inline 01_fibonacci/fib.go]: Nice one-liner. You should use `:=`, though.

File: 01_fibonacci/fib.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fibonacci(10)
+}
+
+func fibonacci(n int) {
+
+	var sum int
```
Comment (anzdaddy) [inline 01_fibonacci/fib.go]: `sum` must start from zero. You can express this intent with `sum := 0`. But actually: a) you should declare it where you set it inside the loop, and b) you don't really need it at all.

File: 01_fibonacci/fib.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	var prev, cur int = 0, 1
+	var fibSeries string
+
+	for i := 1; i <= n; i++ {
+		// fmt.Println(prev)
+		fibSeries += strconv.Itoa(prev) + " "
+		sum = prev + cur
+		prev = cur
+		cur = sum
```
Comment (anzdaddy) [inline 01_fibonacci/fib.go]: You can also do a one-line here: `prev, cur = cur, prev + cur`.

File: 01_fibonacci/fib_test.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote("0 1 1 2 3 5 8 13 21 34 \n")
```
Comment (anzdaddy) [inline 01_fibonacci/fib_test.go]: Not really necessary, but it would be nice to avoid the trailing space.

File: 01_fibonacci/fib.go (hunk 1 above)
Comment (madhavansu) [reply 01_fibonacci/fib.go]: Yes, corrected

File: 01_fibonacci/fib.go (hunk 2 above)
Comment (madhavansu) [reply 01_fibonacci/fib.go]: yes, added as a local variable

File: 01_fibonacci/fib.go (hunk 3 above)
Comment (madhavansu) [reply 01_fibonacci/fib.go]: Yes, correct, but somehow I get weird output if I put one-liner

File: 01_fibonacci/fib_test.go (hunk 4 above)
Comment (madhavansu) [reply 01_fibonacci/fib_test.go]: Yes, removed trailing space

File: 01_fibonacci/fib.go (hunk 3 above)
Comment (juliaogris) [reply 01_fibonacci/fib.go]: This is unnecessarily complicated:
```go
	a, b := 1, 1
	for i := 1; i <= n; i++ {
		fmt.Fprintln(a)
		a, b = b, a+b
	}
```
should do the trick.
If you want to collect values, collect them in a slice and return it. Let main handle the printing....

**Review by juliaogris — COMMENTED** [review]
Please update PR description.
Please edit commit messages according to [recommendations](https://chris.beams.io/posts/git-commit/).

File: 01_fibonacci/fib.go (hunk 3 above)
Comment (anz-rfc) [reply 01_fibonacci/fib.go]: > if you want to collect values, collect them in a slice and return it. Let main handle the printing

i agree! https://github.com/anz-bank/go-training/issues/37

**Comment by paraganz** [comment]
Please add PR description according to [template](https://github.com/anz-bank/go-training/blob/master/.github/pull_request_template.md).

File: 01_fibonacci/fib_test.go (hunk 4 above)
Comment (paraganz) [reply 01_fibonacci/fib_test.go]: Assignments asks to print the numbers on newline and also from 1 i.e.
1
1
2
3
5
8
13

File: 01_fibonacci/fib.go (hunk 3 above)
Comment (madhavansu) [reply 01_fibonacci/fib.go]: Corrected with the right approach.

File: 01_fibonacci/fib_test.go (hunk 4 above)
Comment (madhavansu) [reply 01_fibonacci/fib_test.go]: Done

**Review by juliaogris — COMMENTED** [review]
Almost there, just please update your commits:

    -   Use the imperative mood in the subject line
    -   Fold fixup commits into previous ones with `git rebase -i HASH`
I can give a hand with this if you want

**Comment by madhavansu** [comment]
@juliaogris 
Done with
-   Use the imperative mood in the subject line
-   Fold fixup commits into previous ones with `git rebase -i HASH`

File: 01_fibonacci/fib_test.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote(`1
```
Comment (juliaogris) [inline 01_fibonacci/fib_test.go:21]: Please add a few more testcases n: -1, 0, 1

**Review by juliaogris — COMMENTED** [review]
-   Please edit your commits to follow [recommendations](https://chris.beams.io/posts/git-commit/)
    -   Capitalize the subject line
    -   Use the imperative mood in the subject line
    -   Fold fixup commits into previous ones with `git rebase -i HASH`
I can give a hand with this if you want

**Pull request closed without merging**